owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.kt

```diff
@@ -0,0 +1,437 @@
+        var rootView: RelativeLayout,
+        var textLayout: View,
+        var tabLayout: TabLayout,
+        var viewPager: ViewPager2
+    ) : AsyncTask<OCFile, Unit, StringWriter>() {
+
+        private val DIALOG_WAIT_TAG = "DIALOG_WAIT"
+        private lateinit var mimeType: String
+
+        override fun onPreExecute() {
+            showLoadingDialog()
+        }
+
+        override fun doInBackground(vararg params: OCFile?): StringWriter {
+            if (params.size != 1) {
+                throw IllegalArgumentException("The parameter to ${this.getScopeName()} must be (1) the file location")
+            }
+
+            val file = params[0] as OCFile
+            val location = file.storagePath
+            mimeType = file.mimeType
+
+            var inputStream: FileInputStream? = null
+            var scanner: Scanner? = null
+            val source = StringWriter()
+            val bufferedWriter = BufferedWriter(source)
+
+            try {
+                inputStream = FileInputStream(location)
+                scanner = Scanner(inputStream)
+                while (scanner.hasNextLine()) {
+                    bufferedWriter.append(scanner.nextLine())
+                    if (scanner.hasNextLine()) {
+                        bufferedWriter.append("\n")
+                    }
+                }
+                bufferedWriter.close()
+                val exc = scanner.ioException()
+                if (exc != null) {
+                    throw exc
+                }
+            } catch (e: IOException) {
+                Timber.e(e)
+            } finally {
+                if (inputStream != null) {
+                    try {
+                        inputStream.close()
+                    } catch (e: IOException) {
+                        Timber.e(e)
+                    }
+                }
+                scanner?.close()
+            }
+            return source
+        }
+
+        override fun onPostExecute(result: StringWriter) {
+            val text = String(result.buffer)
+            showPreviewText(text, mimeType, rootView, textViewReference, textLayout, tabLayout, viewPager)
+
+            try {
+                dismissLoadingDialog()
+            } catch (illegalStateException: java.lang.IllegalStateException) {
+                Timber.w(illegalStateException, "Dismissing dialog not allowed after onSaveInstanceState")
+            }
+        }
+
+        fun showLoadingDialog() {
+            val waitDialogFragment = requireActivity().supportFragmentManager.findFragmentByTag(DIALOG_WAIT_TAG)
+            val loading: LoadingDialog
+
+            if (waitDialogFragment == null) {
+                loading = LoadingDialog.newInstance(R.string.wait_a_moment, false)
+                val fragmentManager = requireActivity().supportFragmentManager
+                val fragmentTransaction = fragmentManager.beginTransaction()
+                loading.show(fragmentTransaction, DIALOG_WAIT_TAG)
```

Comment:
Don't we need here a nullable check for `fragmentTransaction` as it was in the legacy code? Just asking

Reply:
Here, it's no necessary because nullable check for `fragmentTransaction` is always true.